packages/datetime/src/timePicker.tsx

```diff
@@ -257,21 +280,25 @@ export class TimePicker extends React.Component<TimePickerProps, ITimePickerStat
     }
 
     private renderInput(className: string, unit: TimeUnit, value: string) {
-        const isValid = isTimeUnitValid(unit, parseInt(value, 10));
+        const valueNumber = parseInt(value, 10);
+        const isValid = isTimeUnitValid(unit, valueNumber);
         const isHour = unit === TimeUnit.HOUR_12 || unit === TimeUnit.HOUR_24;
 
         return (
             <input
+                aria-valuenow={valueNumber}
                 className={classNames(
                     Classes.TIMEPICKER_INPUT,
                     { [CoreClasses.intentClass(Intent.DANGER)]: !isValid },
                     className,
                 )}
+                id={this.timeInputIds[unit]}
                 onBlur={this.getInputBlurHandler(unit)}
                 onChange={this.getInputChangeHandler(unit)}
                 onFocus={this.getInputFocusHandler(unit)}
                 onKeyDown={this.getInputKeyDownHandler(unit)}
                 onKeyUp={this.getInputKeyUpHandler(unit)}
+                role={this.props.showArrowButtons ? "spinbutton" : "textinput"}
```

Comment:
I don't see "textinput" as a valid ARIA role anywhere, do you have documentation for that?

I think we can just leave it undefined when there are no arrow buttons, and inherit the input type="text" role...

```suggestion
                role={this.props.showArrowButtons ? "spinbutton" : undefined}
```